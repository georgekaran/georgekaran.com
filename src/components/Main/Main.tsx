import React from 'react'
import * as S from './Main.styles'

type Props = {
  title?: string
}

const Main: React.FC<Props> = ({
  title = 'Next-Boilerplate'
}: Props) => {
  return (
    <S.Wrapper>
      <S.LogoWrapper>
        <S.Logo src="/img/icon-192.png" alt="Atom image" />
        <S.Title data-testid="title">{title}</S.Title>
      </S.LogoWrapper>
      <S.Description>A simple boilerplate to work with Typescript, React, NextJs and Styled Components</S.Description>
      <S.Illustration src="/img/hero-illustration.svg" alt="Person on a computer" />
    </S.Wrapper>
  )
}

export default Main
