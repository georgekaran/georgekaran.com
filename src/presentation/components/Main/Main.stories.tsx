import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'

import Main from './Main'

export default {
  title: 'Main',
  component: Main
} as Meta

export const Basic: Story = (args) => <Main {...args} />
