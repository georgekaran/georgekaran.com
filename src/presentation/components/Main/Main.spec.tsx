import React from 'react'
import { render, screen, RenderResult } from '@testing-library/react'

import Main from './Main'

const makeSut = (): RenderResult => {
  return render(<Main />)
}

describe('Main component', () => {
  test('Should render with no error', () => {
    makeSut()
    expect(screen.getByTestId('title')).toHaveTextContent('Next-Boilerplate')
  })

  test('Should match snapshot', () => {
    const { container } = makeSut()
    expect(container.firstChild).toMatchSnapshot()
  })
})
