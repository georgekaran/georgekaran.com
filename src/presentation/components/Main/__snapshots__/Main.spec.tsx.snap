// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Main component Should match snapshot 1'] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 2rem;
}

.c2 {
  width: 15rem;
  margin-right: 1rem;
}

.c3 {
  font-size: 4.5rem;
}

.c4 {
  font-size: 2rem;
  font-weight: 400;
}

.c5 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Atom image"
      class="c2"
      src="/img/icon-192.png"
    />
    <h1
      class="c3"
      data-testid="title"
    >
      Next-Boilerplate
    </h1>
  </div>
  <h2
    class="c4"
  >
    A simple boilerplate to work with Typescript, React, NextJs and Styled Components
  </h2>
  <img
    alt="Person on a computer"
    class="c5"
    src="/img/hero-illustration.svg"
  />
</main>
`
